graphql/cxs-impl/src/main/java/org/apache/unomi/graphql/scalars/GeoPointFunction.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.unomi.graphql.scalars;
+
+import graphql.annotations.processor.ProcessingElementsContainer;
+import graphql.annotations.processor.typeFunctions.TypeFunction;
+import graphql.language.StringValue;
+import graphql.schema.Coercing;
+import graphql.schema.CoercingParseLiteralException;
+import graphql.schema.CoercingParseValueException;
+import graphql.schema.CoercingSerializeException;
+import graphql.schema.GraphQLScalarType;
+import graphql.schema.GraphQLType;
+import org.apache.unomi.api.GeoPoint;
+
+import java.lang.reflect.AnnotatedType;
+
+import static graphql.scalars.util.Kit.typeName;
+
+public class GeoPointFunction implements TypeFunction {
+
+    @Override
+    public String getTypeName(Class<?> aClass, AnnotatedType annotatedType) {
+        return GEOPOINT_SCALAR.getName();
+    }
+
+    @Override
+    public boolean canBuildType(Class<?> aClass, AnnotatedType annotatedType) {
+        return aClass == GeoPoint.class;
+    }
+
+    @Override
+    public GraphQLType buildType(boolean input, Class<?> aClass, AnnotatedType annotatedType, ProcessingElementsContainer container) {
+        return GEOPOINT_SCALAR;
+    }
+
+    public static GraphQLScalarType GEOPOINT_SCALAR = GraphQLScalarType.newScalar()
```

Comment:
Will be better to implement this scalar as class which extend `GraphQLScalarType` instead of an instance of class. The last can not be specified as type of list. For example, `List<GEOPOINT_SCALAR>` is not possible to use.